src/solidlsp/language_servers/r_language_server.py

```diff
@@ -0,0 +1,294 @@
+            return
+
+        # Register LSP message handlers
+        self.server.on_request("client/registerCapability", register_capability_handler)
+        self.server.on_notification("window/logMessage", window_log_message)
+        self.server.on_notification("$/progress", do_nothing)
+        self.server.on_notification("textDocument/publishDiagnostics", do_nothing)
+
+        self.logger.log("Starting R Language Server process", logging.INFO)
+        self.server.start()
+
+        initialize_params = self._get_initialize_params(self.repository_root_path)
+        self.logger.log(
+            "Sending initialize request to R Language Server",
+            logging.INFO,
+        )
+
+        init_response = self.server.send.initialize(initialize_params)
+
+        # Verify server capabilities
+        capabilities = init_response.get("capabilities", {})
+        assert "textDocumentSync" in capabilities
+        if "completionProvider" in capabilities:
+            self.logger.log("R LSP completion provider available", logging.INFO)
+        if "definitionProvider" in capabilities:
+            self.logger.log("R LSP definition provider available", logging.INFO)
+
+        self.server.notify.initialized({})
+        self.completions_available.set()
+
+        # R Language Server is ready after initialization
+        self.server_ready.set()
+
+        # Pre-warm the cache by indexing R files to prevent 1+ minute delay on first symbol search
+        self._prewarm_cache()
+
+    @override
+    def request_document_symbols(
+        self, relative_file_path: str, include_body: bool = False
+    ) -> tuple[list[ls_types.UnifiedSymbolInformation], list[ls_types.UnifiedSymbolInformation]]:
+        """
+        Override document symbol request with R-specific error handling.
+
+        The R languageserver can throw "invalid AST" errors (-32001) when parsing
+        certain R files. This method provides graceful fallback to empty results
+        rather than crashing.
+        """
+        try:
+            return super().request_document_symbols(relative_file_path, include_body)
+        except SolidLSPException as e:
+            # Check if this is an AST parsing error (code -32001)
+            if "invalid AST" in str(e).lower() or "-32001" in str(e):
+                self.logger.log(
+                    f"R Language Server AST parsing error for {relative_file_path}: {e}. "
+                    f"This often happens with complex R syntax or when the file has "
+                    f"constructs that the languageserver cannot parse. Trying fallback method.",
+                    logging.WARNING,
+                )
+                # Try fallback method for better symbol extraction
+                return self._extract_r_symbols_fallback(relative_file_path)
+            else:
+                # Re-raise other types of LSP exceptions
+                raise
+
+    def _prewarm_cache(self) -> None:
```

Comment:
Ok, now that CI has passed, I can review the content

This prewarming is not needed and should be removed. Find symbol of only show if no relative paths has been passed to restrict the search. We also have the CLI command project index that fills up the cache and that users can call explicitly